### ISSUE TITLE
Nil pointer dereference during pruning on tx metadata object

### ISSUE THREAD

**Issue by Thoralf-M** [iotaledger/hornet#440]
**Environment information:**
 - OS: Ubuntu 18.04.4 LTS
 - RAM: 4 GB
 - Cores: 2
 - Type: VPS
 - HORNET version [0.4.0-rc9]

https://ybin.me/p/ddf7274faa421325#8XKv9jwDsm1wWC1l3clcRHV+WGcnJO4LoYrv5f6/8oc=
```
May 20 11:30:19 comnetubuntu hornet[21993]:   "snapshots": {
May 20 11:30:19 comnetubuntu hornet[21993]:     "global": {
May 20 11:30:19 comnetubuntu hornet[21993]:       "index": 0,
May 20 11:30:19 comnetubuntu hornet[21993]:       "path": "snapshot.csv",
May 20 11:30:19 comnetubuntu hornet[21993]:       "spentaddressespaths": []
May 20 11:30:19 comnetubuntu hornet[21993]:     },
May 20 11:30:19 comnetubuntu hornet[21993]:     "loadtype": "local",
May 20 11:30:19 comnetubuntu hornet[21993]:     "local": {
May 20 11:30:19 comnetubuntu hornet[21993]:       "depth": 20,
May 20 11:30:19 comnetubuntu hornet[21993]:       "downloadurl": "https://ls.manapotion.io/comnet/export.bin",
May 20 11:30:19 comnetubuntu hornet[21993]:       "intervalsynced": 50,
May 20 11:30:19 comnetubuntu hornet[21993]:       "intervalunsynced": 1000,
May 20 11:30:19 comnetubuntu hornet[21993]:       "path": "export_comnet.bin"
May 20 11:30:19 comnetubuntu hornet[21993]:     },
May 20 11:30:19 comnetubuntu hornet[21993]:     "pruning": {
May 20 11:30:19 comnetubuntu hornet[21993]:       "delay": 50,
May 20 11:30:19 comnetubuntu hornet[21993]:       "enabled": true
May 20 11:30:19 comnetubuntu hornet[21993]:     }
May 20 11:30:19 comnetubuntu hornet[21993]:   },
```

**Comment by luca-moser** [comment]
After starting a node with pruning set to 1k and letting it prune data, the node crashed as a tx metadata object was nil.

https://ybin.me/p/064f5c76e4a91b5b#ytpu/rxfZ23Tx8UBAv8y//Ayc4m7xOTmVe++8gHJjoc=

**Comment by centercirclesolutions** [comment]
Similar with Pruning set to 150
CX41 Version 0.4.0-rc10-7511e71


[Log](https://ybin.me/p/e4d493e10d683cec#cBF9DdenM+gf/4n1ePriV6ho0u/fqn8WXvXNAF2KL30=)

**Comment by alexsporn** [comment]
Fixed with #450
